src/en/clowncorps/src/eu/kanade/tachiyomi/extension/en/clowncorps/ClownCorps.kt

```diff
@@ -0,0 +1,183 @@
+        "It's pronounced \"core.\" Like marine corps."
+
+    override val client = network.client.newBuilder()
+        .addInterceptor(TextInterceptor())
+        .build()
+
+    override fun fetchPopularManga(page: Int): Observable<MangasPage> = Observable.fromCallable {
+        MangasPage(
+            listOf(
+                SManga.create().apply {
+                    title = name
+                    artist = creator
+                    author = creator
+                    description = synopsis
+                    status = SManga.ONGOING
+                    // Image from: https://clowncorps.net/about/
+                    thumbnail_url = "https://clowncorps.net/wp-content/uploads/2022/11/clowns41.jpg"
+                    setUrlWithoutDomain("/comic")
+                },
+            ),
+            hasNextPage = false,
+        )
+    }
+
+    override fun fetchSearchManga(page: Int, query: String, filters: FilterList) =
+        fetchPopularManga(page)
+
+    override fun fetchMangaDetails(manga: SManga): Observable<SManga> =
+        Observable.just(manga.apply { initialized = true })
+
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val pageStatus = response.asJsoup().select("#paginav li.paginav-pages").text()
+        val pageCount = pageStatus.split(" ").last().toInt()
+
+        val chapters = mutableListOf<SChapter>()
+
+        for (page in 1..pageCount) {
+            val url = "https://clowncorps.net/comic/page/$page/"
+            val resp = client.newCall(GET(url, headers)).execute()
+            if (resp.code == 404) break
+            val comics = resp.asJsoup().select(".comic")
+            for (comic in comics) {
+                val link = comic.selectFirst(".post-title a")?.attr("href")
+                val title = comic.selectFirst(".post-title a")?.text()
+                val postDate = comic.selectFirst(".post-date")?.text()
+                val postTime = comic.selectFirst(".post-time")?.text()
+                if (link == null || title == null || postDate == null || postTime == null) continue
+                val date = parseDate("$postDate $postTime")
+
+                val chapter = SChapter.create().apply {
+                    setUrlWithoutDomain(link)
+                    name = title
+                    date_upload = date
+                }
+                chapters.add(chapter)
+            }
+        }
+
+        // Add chapter numbers to the chapters, to ensure correct sorting.
+        // Seems to be unnecessary, as it already bases the order on the chapter names.
+        // Kept here, commented out, just in case.
+        /*chapters.forEachIndexed { index, sChapter ->
+            sChapter.chapter_number = chapters.size - index.toFloat()
+        }*/
+
+        return chapters
+    }
+
+    private fun parseDate(dateStr: String): Long {
+        return try {
+            dateFormat.parse(dateStr)!!.time
```

Comment:
```suggestion
            dateFormat.parse(dateStr)?.time ?: 0L
```

Reply:
@AwkwardPeak7 No it won't be `null` at all, see [`Date parse(String) throws ParseException`](https://docs.oracle.com/javase/8/docs/api/java/text/DateFormat.html#parse-java.lang.String-). IDE wrongly prompts it as nullable because it wraps [`Date? parse(String, ParsePosition)`](https://docs.oracle.com/javase/8/docs/api/java/text/DateFormat.html#parse-java.lang.String-java.text.ParsePosition-) and throws on a check of `ParsePosition` so the IDE can't tell.